langchain/schema.py

```diff
@@ -266,17 +330,50 @@ def __eq__(self, other: object) -> bool:
 
 
 class PromptValue(Serializable, ABC):
+    """Base abstract class for inputs to any language model.
+
+    PromptValues can be converted to both LLM (pure text-generation) inputs and
+        ChatModel inputs.
+    """
+
     @abstractmethod
     def to_string(self) -> str:
-        """Return prompt as string."""
+        """Return prompt value as string."""
 
     @abstractmethod
     def to_messages(self) -> List[BaseMessage]:
-        """Return prompt as messages."""
+        """Return prompt as a list of Messages."""
 
 
 class BaseMemory(Serializable, ABC):
-    """Base interface for memory in chains."""
+    """Base abstract class for memory in Chains.
+
+    Memory refers to state in Chains. Memory can be used to store information about
+        past executions of a Chain and inject that information into the inputs of
+        future executions of the Chain. For example, for conversational Chains Memory
+        can be used to store conversations and automatically add them to future model
+        prompts so that the model has the necessary context to respond coherently to
+        the latest input.
+
+    Example:
+        .. code-block:: python
+
+            class SimpleMemory(BaseMemory):
```

Comment:
does this class actually exist? 

I can see this being a bit confusing and I would maybe remove, im not sure this is conventional